File: src/bytecode/structs/opcodes/OpLe.ts
import { Instruction } from "../Instruction";
import { Opcode } from "../Opcode";

export class OpLe extends Opcode {
  instruction: Instruction;

  constructor(instruction: Instruction) {
    super();

    this.instruction = instruction;
  }

  getObfuscated(): string {
    throw new Error("Method not implemented.");
  }
}

File: src/bytecode/structs/opcodes/op_le.rs
use crate::bytecode::structs::{instruction::Instruction, opcode::Opcode};

pub struct OpLe {
    instruction: Instruction,
}

impl OpLe {
    pub fn new(instruction: Instruction) -> Self {
        Self { instruction }
    }
}

impl Opcode for OpLe {
    fn get_instruction(&self) -> &Instruction {
        &self.instruction
    }

    fn get_obfuscated(&self) -> &str {
        "if (constantB(inst) <= constantC(inst)) == (inst.A ~= 0) then pc = pc + code[pc].sBx end

        pc = pc + 1"
    }

    fn is_valid(&self) -> bool {
        todo!()
    }
}
